rinad/src/ipcm/ipcm.cc

```diff
@@ -395,6 +400,26 @@ IPCManager_::assign_to_dif(Addon* callee, Promise* promise,
 	IPCPTransState* trans;
 
 	try {
+		if (is_any_ipcp_assigned_to_dif(dif_name)) {
+			ss << "There is already an IPCP assigned to DIF "
+				<< dif_name.toString()
+				<< " in this system.";
+			FLUSH_LOG(ERR, ss);
+			throw rina::AssignToDIFException();
+		}
+
+		ipcp = lookup_ipcp_by_id(ipcp_id, false);
+		if (ipcp->get_type() == rina::NORMAL_IPC_PROCESS) {
+			// Load all the plugins required from by template, but
+			// first release the ipcp lock, to avoid lock-ups.
+			ipcp->rwlock.unlock();
+
+			catalog.load_by_template(callee, ipcp_id, dif_template);
```

Comment:
Self-ACK or partial-ACK it's against our rules, I won't do it.

All right, I guess who needs this (e.g. we need it in WP4/LFA) can merge it
in a local branch and work there for experimentation.

I'll try to consider your concerns when I've time - unless you want to fix
it yourself, in which case would be ok for me.

2015-07-17 18:02 GMT+02:00 Marc Suñé <notifications@github.com>:

> In rinad/src/ipcm/ipcm.cc
> <https://github.com/IRATI/stack/pull/678#discussion_r34903550>:
>
> > @@ -395,6 +400,26 @@ IPCManager_::assign_to_dif(Addon* callee, Promise* promise,
> >  	IPCPTransState* trans;
> >
> >  	try {
> > +		if (is_any_ipcp_assigned_to_dif(dif_name)) {
> > +			ss << "There is already an IPCP assigned to DIF "
> > +				<< dif_name.toString()
> > +				<< " in this system.";
> > +			FLUSH_LOG(ERR, ss);
> > +			throw rina::AssignToDIFException();
> > +		}
> > +
> > +		ipcp = lookup_ipcp_by_id(ipcp_id, false);
> > +		if (ipcp->get_type() == rina::NORMAL_IPC_PROCESS) {
> > +			// Load all the plugins required from by template, but
> > +			// first release the ipcp lock, to avoid lock-ups.
> > +			ipcp->rwlock.unlock();
> > +
> > +			catalog.load_by_template(callee, ipcp_id, dif_template);
>
> That's a one hour rework (or less).
>
> But if you both think it is so urgent it cannot wait, just ACK it
> yourselves and merge it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IRATI/stack/pull/678/files#r34903550>.
>



-- 
Vincenzo Maffione